test/e2e/kubectl.go

```diff
@@ -125,7 +125,7 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 	framework.KubeDescribe("Update Demo", func() {
 		var updateDemoRoot, nautilusPath, kittenPath string
 		BeforeEach(func() {
-			updateDemoRoot = filepath.Join(framework.TestContext.RepoRoot, "docs/user-guide/update-demo")
+			updateDemoRoot = filepath.Join(framework.TestContext.GeneratedAssetsDir, framework.GENERATED_ASSETS, "docs/user-guide/update-demo")
```

Comment:
generated assets defines a directory where the test writes it's files to, using Asset like you described.

The support nightmare that your referring to is exactly why we need go bindata, swapping repo-root or providing it at runtime is ugly ... so I'm pretty sure we agree :)